smtp.go

```diff
@@ -95,8 +103,16 @@ func (d *Dialer) Dial() (SendCloser, error) {
 		}
 	}
 
-	if !d.SSL {
-		if ok, _ := c.Extension("STARTTLS"); ok {
+	if !d.SSL && d.RequireStartTLS != NoStartTLS {
+		ok, _ := c.Extension("STARTTLS")
+		if !ok && d.RequireStartTLS == MandatoryStartTLS {
+			err := fmt.Errorf(
```

Comment:
We should export an error for this (e.g. `ErrStartTLSUnsupported`).

Reply:
Fixed in d3eed43, but named `StartTLSUnsupportedError` to follow Go conventions: https://blog.golang.org/error-handling-and-go

Reply:
That much better than my suggestion. I like your use of `StartTLSPolicy.String()` for handling the error message.😄